server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/vacation/VacationResponseSet.scala

```diff
@@ -0,0 +1,113 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *  http://www.apache.org/licenses/LICENSE-2.0                  *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.vacation
+
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+import org.apache.james.jmap.api.vacation.VacationPatch
+import org.apache.james.jmap.mail.MailboxSetError.{invalidArgumentValue, serverFailValue}
+import org.apache.james.jmap.mail.{SetErrorDescription, SetErrorType}
+import org.apache.james.jmap.model.AccountId
+import org.apache.james.jmap.model.State.State
+import org.apache.james.util.ValuePatch
+import play.api.libs.json.{JsBoolean, JsNull, JsObject, JsString, JsValue}
+
+case class VacationResponseSetRequest(accountId: AccountId,
+                                      update: Option[Map[String, VacationResponsePatchObject]],
+                                      create: Option[Map[String, JsObject]],
+                                      destroy: Option[Seq[String]]) {
+  def parsePatch(): Map[String, Either[IllegalArgumentException, VacationResponsePatchObject]] =
+    update.getOrElse(Map())
+    .map({
+      case (id, patch) if id.equals("singleton") => (id, Right(patch))
```

Comment:
I seen `"singleton"` as constant somewhere, we can reused it or make own constant in this file.

Reply:
> > Oups we have a timezone issue!
> 
> Ah then the Serializer method doesn't work ! Maybe you should keep the one I did with the get response? :)

Yep, should have known better :)